js/util/polygon_poi.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+var Queue = require('tinyqueue');
+var Point = require('point-geometry');
+var distToSegmentSquared = require('./intersection_tests').distToSegmentSquared;
+var isClosedPolygon = require('./util').isClosedPolygon;
+
+/**
+ * Finds an approximation of a polygon's Pole Of Inaccessibiliy https://en.wikipedia.org/wiki/Pole_of_inaccessibility
+ * Note that for perfect rectangles the pole found ISN'T in the center (though the center is a valid POI)
```

Comment:
The note about perfect rectangles is no longer valid since we get centroid as the first best guess and only choose a different point if it's better in terms of distance. 

I think it's also worth linking to the original repo (https://github.com/mapbox/polylabel) in the comments.